File: Sep19/Lecture8Examples/Example32/NotificationJobService.java
package com.varunshankar.example32;

import android.app.job.JobParameters;
import android.app.job.JobService;

import androidx.core.app.NotificationCompat;
import android.widget.Toast;

public class NotificationJobService extends JobService {
    @Override
    public boolean onStartJob(JobParameters jobParameters) {
        //Show a toast
        Toast.makeText(getApplicationContext(),"Job is now running",Toast.LENGTH_SHORT).show();
        jobFinished(jobParameters,false);
        return false;
    }

    @Override
    public boolean onStopJob(JobParameters jobParameters) {
        return true;
    }
}

File: Sep19/Example32/app/src/main/java/com/varunshankar/example32/NotificationJobService.kt
package com.varunshankar.example32

import android.app.job.JobService
import android.app.job.JobParameters
import android.widget.Toast

class NotificationJobService : JobService() {
    override fun onStartJob(jobParameters: JobParameters): Boolean {
        //Show a toast
        Toast.makeText(applicationContext, "Job is now running", Toast.LENGTH_SHORT).show()
        jobFinished(jobParameters, false)
        return false
    }

    override fun onStopJob(jobParameters: JobParameters): Boolean {
        return true
    }
}